src/somnium/congomongo.clj

```diff
@@ -412,8 +412,8 @@ When with-mongo and set-connection! interact, last one wins"
   (when (and one? sort)
     (throw (IllegalArgumentException. "Fetch :one? (or fetch-one) can't be used with :sort.
 You should use fetch with :limit 1 instead."))); one? and sort should NEVER be called together
-  (when (and one? (or read-preferences (not= [] options) (not= 0 limit) hint explain?))
-    (throw (IllegalArgumentException. "At the moment, fetch-one doesn't support read-preferences, options, limit or hint"))) ;; these are allowed but not implemented here
+  (when (and one? (or (not= [] options) (not= 0 limit) hint explain?))
+    (throw (IllegalArgumentException. "At the moment, fetch-one doesn't support options, limit, or hint"))) ;; these are allowed but not implemented here
```

Comment:
I'll leave as-is, the stack-trace includes where the exception comes from. Exceptions thrown by the driver come from `com.mongodb`, e.g. https://rollbar.com/circle/circle/items/27650/